apps/omg_watcher/lib/omg_watcher/exit_processor.ex

```diff
@@ -268,64 +269,46 @@ defmodule OMG.Watcher.ExitProcessor do
 
   def handle_call({:finalize_in_flight_exits, finalizations}, _from, state) do
     _ = if not Enum.empty?(finalizations), do: Logger.info("Recognized ife finalizations: #{inspect(finalizations)}")
-
-    case Core.finalize_in_flight_exits(state, finalizations) do
-      {:ok, state, db_updates} ->
-        {:reply, {:ok, db_updates}, state}
-
-      {:unknown_piggybacks, unknown_piggybacks} ->
-        _ = Logger.error("Outputs not piggybacked: #{inspect(unknown_piggybacks)}")
-        {:stop, :unknown_piggybacks, state}
-
-      {:unknown_in_flight_exit, unknown_ifes} ->
-        _ = Logger.error("Unknown in-flight exits: #{inspect(unknown_ifes)}")
-        {:stop, :unknown_in_flight_exit, Elixir.Agent.Server, state}
-    end
+    {:ok, state, db_updates} = Core.finalize_in_flight_exits(state, finalizations)
+    {:reply, {:ok, db_updates}, state}
   end
 
   @doc """
   Combine data from `ExitProcessor` and `OMG.State` to figure out what to do about exits
   """
   def handle_call(:check_validity, _from, state) do
```

Comment:
I am not sure whether the atom `:check_validity` is relevant to what the handler does

Reply:
it is `GenServer.called` by `check_validity/0` above and it calls `Core.check_validity`.

When it comes to the name itself - the call in general checks the validity of the exits - it checks standard and in-flight exits for all possible kinds of "invalidities" (aka `byzantine_events`).

`check_validity` looks like a reasonable name to me, but do we have any alternative proposition?

Reply:
You are right, but the name suggests that the function might be pure (just checks the validity), but it is not that important so we wanted to waste time on it :wink:

Reply:
oh, but this is a v. good point! To an external user, `check_validity` is what is being done. It's still "pure" in the sense that the stuff that is non-pure is just caching being done.

I'll keep the name, but put docs in stating, that this f might also do some amount of caching for subsequent calls